onnxruntime/python/tools/transformers/fusion_embedlayer.py

```diff
@@ -53,22 +55,34 @@ def fuse(self, node, input_name_to_nodes, output_name_to_node):
                 return
             children = input_name_to_nodes[node.output[0]]
             children_types = sorted([child.op_type for child in children])
-            if children_types != ['MatMul', 'MatMul', 'MatMul', 'SkipLayerNormalization']:
+            if children_types != ['MatMul', 'MatMul', 'MatMul', 'SkipLayerNormalization'] and children_types != ['MatMul', 'MatMul', 'MatMul', 'Shape', 'Shape', 'SkipLayerNormalization']:
                 logger.debug("No Attention like subgraph in children of SkipLayerNormalization")
                 return
 
         # Assume the order of embeddings are word_embedding + position_embedding + segment_embedding
         normalize_node = node
+        add_node = None
         word_embedding_path = self.model.match_parent_path(normalize_node, ['Add', 'Gather'], [0, 0])
-        if word_embedding_path is None:
-            logger.info("Word embedding path is not found. Embed layer cannot be fused.")
-            return
-        add_node, word_embedding_gather = word_embedding_path
+        if word_embedding_path is not None:
+            add_node, word_embedding_gather = word_embedding_path
+        else:
+            word_embedding_path = self.model.match_parent_path(normalize_node, ['Gather'], [0])
+            if word_embedding_path is not None:
+                word_embedding_path = self.model.match_parent_path(normalize_node, ['Gather'], [0])
+                word_embedding_gather = word_embedding_path[0]
+                is_distill = True;
+            else:
+                logger.info("Word embedding path is not found. Embed layer cannot be fused.")
+                return
+
         input_ids = word_embedding_gather.input[1]
 
         position_embedding_expand = None
         position_embedding_shape = None
 
+        if is_distill:
+            add_node = normalize_node
```

Comment:
This is confusing. add_node is None, why assign normlize_node to it.

Reply:
it was initially designed for reusing line 99. But I'll refactor them to avoid confusing